dlt/cli/magics.py

```diff
@@ -0,0 +1,286 @@
+    """
+    if is_ipython():
+        try:
+            register_magics()
+        except ImportError:
+            pass
+
+# type: ignore
+from IPython.core.display import HTML, display # type: ignore
+from IPython.core.magic import ( Magics, cell_magic,line_cell_magic,line_magic, magics_class, register_line_magic,)
+from IPython.core.magic_arguments import argument, magic_arguments, parse_argstring
+
+
+@magics_class
+class DltMagics(Magics):
+
+    @property
+    def display(self) ->  t.Any:
+        if os.getenv("DATABRICKS_RUNTIME_VERSION"):
+            # Assume Databricks' 'display' is a callable with an unknown signature
+            databricks_display = self._shell.user_ns.get("display")
+            if callable(databricks_display):
+                return databricks_display
+            else:
+                raise RuntimeError("Expected a callable for Databricks' display, got something else.")
+        return display  # Assuming 'display' is a predefined callable
+
+    def success_message(self, messages: t.Dict[str, str]) -> HTML:
+        unstyled = messages.get("unstyled")
+        msg = str(
+            h(
+                "div",
+                h(
+                    "span",
+                    messages.get("green-bold"),
+                    {"style": {"color": "green", "font-weight": "bold"}},
+                ),
+                h("span", unstyled) if unstyled else "",
+            )
+        )
+        return HTML(msg)
+
+    def on_exception(self, ex: str, info: str) -> t.Any:
+        msg = str(
+            h(
+                "div",
+                h(
+                    "span",
+                    str(ex),
+                    {"style": {"color": "red", "font-weight": "bold"}},
+                ),
+                h(
+                    "span",
+                    " Please refer to %s for further assistance" % fmt.bold(info)
+                ),
+            )
+        )
+        out = display(HTML(msg))
+
+        if DEBUG_FLAG:
+            # Display the full traceback in a preformatted style
+            traceback_html = h("pre", traceback.format_exc(), {"style": {"color": "gray"}})
+            display(HTML(str(traceback_html)))
+            raise ex
+
+        return out
+
+
+    @magic_arguments()
+    @argument('--enable-telemetry', action='store_true', help="Enables telemetry before command is executed")
+    @argument('--disable-telemetry', action='store_true', help="Disables telemetry before command is executed")
+    @argument('--non-interactive', action='store_true', help="Non interactive mode. Default choices are automatically made for confirmations and prompts.")
+    @argument('--debug', action='store_true', help="Displays full stack traces on exceptions.")
+    @line_magic
+    @register_line_magic
+    def settings(self, line:str)->str:
+        """
+        A DLT line magic command to set global settings like telemetry, debug mode, etc.
```

Comment:
we do not capitalize `dlt`. I know you can find wrong capitalization in the code. let's fix them when we see them